packages/cli/cloud/src/link/action.ts

```diff
@@ -0,0 +1,134 @@
+import inquirer from 'inquirer';
+import chalk from 'chalk';
+
+import type { CLIContext } from '../types';
+import { LocalSave } from '../services/strapi-info-save';
+import { cloudApiFactory, tokenServiceFactory, local } from '../services';
+import { promptLogin } from '../login/action';
+
+export default async (ctx: CLIContext) => {
+  const { getValidToken } = await tokenServiceFactory(ctx);
+  const token = await getValidToken(ctx, promptLogin);
+  const { logger } = ctx;
+
+  if (!token) {
+    return;
+  }
+
+  let existingConfig: LocalSave;
+
+  try {
+    existingConfig = await local.retrieve();
+    if (existingConfig.project) {
+      const { shouldRelink } = await inquirer.prompt([
+        {
+          type: 'confirm',
+          name: 'shouldRelink',
+          message: `A project named ${chalk.cyan(
+            existingConfig.project.displayName
+              ? existingConfig.project.displayName
+              : existingConfig.project.name
+          )} is already linked to this local folder. Do you want to update the link?`,
+          default: false,
+        },
+      ]);
+
+      if (!shouldRelink) {
+        return;
+      }
+    }
+  } catch (e) {
+    logger.debug('Failed to check project config', e);
+    logger.error('An error occurred while trying to link the project.');
+  }
+
+  const cloudApiService = await cloudApiFactory(ctx, token);
+
+  try {
+    await cloudApiService.track('willLinkProject', {});
+  } catch (e) {
+    ctx.logger.debug('Failed to track willLinkProject', e);
+  }
```

Comment:
Maybe we can have an util function to do this, as we do it everywhere.
Like
```javascript
function trackEvent({ logger, cloudApiService }, event, data = {}) {
  try {
    await cloudApiService.track(event, data);
  } catch (e) {
    logger.debug(`Failed to track ${event}`, e);
  }
}
```
WDYT?

Reply:
I though about it, yes. We can have that.